cmd/collector/app/metrics.go

```diff
@@ -136,18 +138,26 @@ func newMetricsBySvc(factory metrics.Factory, category string) metricsBySvc {
 	spansFactory := factory.Namespace(metrics.NSOptions{Name: "spans", Tags: nil})
 	tracesFactory := factory.Namespace(metrics.NSOptions{Name: "traces", Tags: nil})
 	return metricsBySvc{
-		spans:  newCountsBySvc(spansFactory, category, maxServiceNames),
-		traces: newCountsBySvc(tracesFactory, category, maxServiceNames),
+		spans:  newCountsBySvc(spansFactory, category, maxServiceNames, false),
+		traces: newCountsBySvc(tracesFactory, category, maxServiceNames, true),
 	}
 }
 
-func newCountsBySvc(factory metrics.Factory, category string, maxServiceNames int) countsBySvc {
+func newCountsBySvc(factory metrics.Factory, category string, maxServiceNames int, isTraceMetric bool) countsBySvc {
+	countsTags := map[string]string{"svc": otherServices, "debug": "false"}
+	debugCountsTags := map[string]string{"svc": otherServices, "debug": "true"}
+	if isTraceMetric {
+		// The `samplerTypeKey` tag needs be added to otherServices trace metric counter so prometheus
+		// won't panic because `otherServices` share different tags with normal service.
+		countsTags[samplerTypeKey] = "unknown"
```

Comment:
does this mean "other" services will always be classified with the "unknown" `sampler_type`? I'd prefer to have a counter for each `sampler_type` for "unknown" as well if possible.

Reply:
Would this break jaeger users's metric dashboard though? I'm fine with the updates but want to make sure impact is minimum for now.

Reply:
_should_ have no impact